repos/fdbt-site/src/pages/capConfirmation.tsx

```diff
@@ -0,0 +1,229 @@
+import {
+    TYPE_OF_CAP_ATTRIBUTE,
+    CAPS_ATTRIBUTE,
+    CAPPED_PRODUCT_GROUP_ID_ATTRIBUTE,
+    CAP_EXPIRY_ATTRIBUTE,
+    CAP_START_ATTRIBUTE,
+    SERVICE_LIST_ATTRIBUTE,
+    MULTI_TAPS_PRICING_ATTRIBUTE,
+    CAP_PRICING_PER_DISTANCE_ATTRIBUTE,
+} from '../constants/attributes';
+import { NextPageContextWithSession, ConfirmationElement, Cap } from '../interfaces';
+import TwoThirdsLayout from '../layout/Layout';
+import CsrfForm from '../components/CsrfForm';
+import ConfirmationTable from '../components/ConfirmationTable';
+import { getSessionAttribute } from '../utils/sessions';
+import { isCapExpiry, isCapStartInfo, isWithErrors } from '../interfaces/typeGuards';
+import { getCsrfToken, sentenceCaseString, getAndValidateNoc } from '../utils';
+import { getProductGroupByNocAndId } from '../data/auroradb';
+import { CapStartInfo } from 'src/interfaces/matchingJsonTypes';
+import { isServiceListAttributeWithErrors } from './serviceList';
+
+const title = 'Cap Confirmation - Create Fares Data Service';
+const description = 'Cap Confirmation page of the Create Fares Data Service';
+
+interface CapConfirmationProps {
+    typeOfCap: string;
+    productGroupName: string;
+    caps: Cap[];
+    capValidity: string;
+    capStartInfo: CapStartInfo;
+    services: string[];
+    tapsPricingContents: string[];
+    capDistancePricingContents: string[];
+    csrfToken: string;
+}
+
+export const buildCapConfirmationElements = (
+    typeOfCap: string,
+    productGroupName: string,
+    caps: Cap[],
+    capValidity: string,
+    capStartInfo: CapStartInfo,
+    services: string[],
+    tapsPricingContents: string[],
+    capDistancePricingContents: string[],
+): ConfirmationElement[] => {
+    const confirmationElements: ConfirmationElement[] = [
+        {
+            name: 'Cap type',
+            content: `Pricing by ${typeOfCap.substring(2)} `,
+            href: 'typeOfCap',
+        },
+    ];
+
+    if (productGroupName) {
+        confirmationElements.push({
+            name: 'Product Group Name',
+            content: productGroupName,
+            href: '/selectCappedProductGroup',
+        });
+    }
+
+    caps.forEach((cap) => {
+        const durationText = cap.durationUnits
+            ? `${cap.durationAmount} ${cap.durationUnits}${cap.durationAmount === '1' ? '' : 's'}`
+            : `${cap.durationAmount}`;
+
+        confirmationElements.push({
+            name: cap.name,
+            content: [`Price - £${cap.price}`, `Duration - ${durationText}`],
+            href: '/createCaps',
+        });
+    });
+
+    confirmationElements.push({
+        name: 'Cap Expiry',
+        content: sentenceCaseString(capValidity),
+        href: '/selectCapValidity',
+    });
+
+    confirmationElements.push({
+        name: 'Cap Starts',
+        content:
+            capStartInfo.type === 'fixedWeekdays' && capStartInfo.startDay
+                ? `Fixed Days - ${sentenceCaseString(capStartInfo.startDay)}`
```

Comment:
Fixed days
Rolling days

We use sentence case where we can

Reply:
these are hard coded, do we still need the function, i think, better to use when we have dynamic text